plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineManagerImpl.java

```diff
@@ -42,26 +52,69 @@
 @Singleton
 public class MachineManagerImpl implements MachineManager {
 
-    private final MachineServiceClient    machineServiceClient;
-    private final WorkspaceServiceClient  workspaceServiceClient;
-    private final AppContext              appContext;
-    private final DtoFactory              dtoFactory;
-    private final EventBus                eventBus;
+    private final MachineServiceClient   machineServiceClient;
+    private final WorkspaceServiceClient workspaceServiceClient;
+    private final AppContext             appContext;
+    private final EntityFactory          entityFactory;
+    private final DtoFactory             dtoFactory;
+    private final EventBus               eventBus;
 
     @Inject
     public MachineManagerImpl(final MachineServiceClient machineServiceClient,
                               final WorkspaceServiceClient workspaceServiceClient,
                               final EventBus eventBus,
                               final AppContext appContext,
+                              final EntityFactory entityFactory,
                               final DtoFactory dtoFactory) {
         this.machineServiceClient = machineServiceClient;
         this.workspaceServiceClient = workspaceServiceClient;
         this.appContext = appContext;
+        this.entityFactory = entityFactory;
         this.dtoFactory = dtoFactory;
         this.eventBus = eventBus;
     }
 
     @Override
+    public Promise<MachineEntity> getMachine(final String workspaceId, final String machineId) {
+        return workspaceServiceClient.getWorkspace(workspaceId).then(new Function<WorkspaceDto, MachineEntity>() {
+            @Override
+            public MachineEntity apply(WorkspaceDto workspace) throws FunctionException {
+                WorkspaceRuntimeDto workspaceRuntime = workspace.getRuntime();
+                if (workspaceRuntime == null) {
+                    return null;
```

Comment:
_null_ is returning here but javadoc for the _MachineManager#getMachine_ method says nothing about that
So it's potential NPE in _JavaDebugConfigurationPagePresenter#extractPortsList_ method.